### PR TITLE
(fix) Update item date_mod

### PR DESCRIPTION
**Pull request by Lainow**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !33527

After modifying additional fields, the modification date is not updated.

### CODE REVIEW

**Review by stonebuzz — CHANGES_REQUESTED** [review]

File: inc/container.class.php [hunk 1]
```diff
@@ -1649,6 +1649,7 @@ public static function preItemUpdate(CommonDBTM $item)
                 count($data) == 0
                 || $container->updateFieldsValues($data, $item->getType(), isset($_REQUEST['massiveaction']))
             ) {
+                $item->input['date_mod'] = date("Y-m-d H:i:s");
```
Comment (stonebuzz) [inline inc/container.class.php:1652]:
```suggestion
                $item->input['date_mod'] = $_SESSION["glpi_currenttime"];
```

**Review by stonebuzz — CHANGES_REQUESTED** [review]

**Review by AdrienClairembault — COMMENTED** [review]
This is weird because GLPI core is already supposed to do it:

image

**Comment by stonebuzz** [comment]
@Lainow can you complete the CHANGELOG ?

**Comment by stonebuzz** [comment]
> This is weird because GLPI core is already supposed to do it:
> 
> image

only for objects which have a 'date_mod' field and which are affected by the update.

In this case, it is the data from the fields plugin that is updated and not the linked GLPI object (so ```$this->updates``` contains nothing for the GLPI object).

**Comment by AdrienClairembault** [comment]
> > This is weird because GLPI core is already supposed to do it:
> > image
> 
> only for objects which have a `date_mod` field and which are affected by the update.
> 
> In this case, it is the data from the fields plugin that is updated and not the linked GLPI object (so `$this->updates` contains nothing for the GLPI object).

I see, thanks for this details.
I've actually had the same issue on forms when updating a linked question and used the same solution so ok for me 👍.

**Review by AdrienClairembault — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [unrelease] -
+
+### Fixed
+
+- Update item date_mod after updating additional fields
```
Comment (stonebuzz) [inline CHANGELOG.md:12]:
```suggestion
- Update main item ```date_mod``` after updating additional fields
```

**Review by stonebuzz — CHANGES_REQUESTED** [review]

**Review by stonebuzz — APPROVED** [review]

**Pull request merged**